### PR TITLE
Ability to run tests in D/S

### PR DESCRIPTION
**Pull request by mor120**
**What this PR does / why we need it**:
This PR will give us the ability to run tests in a D/S environment using the existing scripts.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
In D/S we are working to run tests in containers instead of running them on VMs (executors).
We can't run docker in docker in D/S in order to build the tests so we are going to use the builder image directly and then copy the binaries to a container that will run the tests.

**Special notes for your reviewer**:
None

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/build/build-functest.sh [hunk 1]
```diff
@@ -14,11 +14,11 @@
...
 set -euo pipefail
 
-script_dir="$(cd "$(dirname "$0")" && pwd -P)"
+script_dir=$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")
 source "${script_dir}"/common.sh
 
 mkdir -p ${TESTS_OUT_DIR}/
 # use vendor
-export GO111MODULE=off
+export ${GO111MODULE:-off}
```
Comment (lukas-bednar) [inline hack/build/build-functest.sh:22]: Shouldn't it be like the following?

```
export GO111MODULE=${GO111MODULE:-off}
```

**Comment by maya-r** [comment]
Can you explain why script_dir is changed? we're avoiding readlink -f because of https://github.com/kubevirt/containerized-data-importer/issues/1317

/ok-to-test

File: hack/build/build-functest.sh (hunk 1 above)
Comment (mor120) [reply hack/build/build-functest.sh:22]: Totally, thanks!

**Comment by mor120** [comment]
> Can you explain why script_dir is changed? we're avoiding readlink -f because of #1317
> 
> /ok-to-test

I don't really need it, I have reverted these changes, thanks :)

**Comment by maya-r** [comment]
this looks good to me -- the bot is complaining about the lack of a "DCO signoff", the lack of a Signed-off-by: author text in the commit message. You can add it to a commit with `git commit --amend --sign`

**Review by akalenyu — COMMENTED** [review]
I am probably cheating here because I looked at the downstream PRs, but I think it would be much better to have a make target for your use case that spares you any extras you have to do yourself.

@maya-r This is about containerizing the test runs of CDI without the ability to do docker-in-docker. So all we get is a worker pod in a cluster span up from the builder image

**Pull request merged**